arch/arm/boot/dts/overlays/revpi-flat-s-2022-overlay.dts

```diff
@@ -0,0 +1,350 @@
+					linux,default-trigger = "power_red";
+				};
+				a1_green {
+					gpios = <&expander 0 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a1_green";
+				};
+				a1_red {
+					gpios = <&expander 1 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a1_red";
+				};
+				a2_green {
+					gpios = <&expander 2 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a2_green";
+				};
+				a2_red {
+					gpios = <&expander 3 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a2_red";
+				};
+				a3_green {
+					gpios = <&expander 4 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a3_green";
+				};
+				a3_red {
+					gpios = <&expander 5 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a3_red";
+				};
+				a4_green {
+					gpios = <&expander 6 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a4_green";
+				};
+				a4_red {
+					gpios = <&expander 7 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a4_red";
+				};
+				a5_green {
+					gpios = <&expander 8 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a5_green";
+				};
+				a5_red {
+					gpios = <&expander 9 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a5_red";
+				};
+				led-act {
+					status = "disabled";
+				};
+			};
+
+			gpio-keys {
+				compatible = "gpio-keys";
+				pinctrl-names = "default";
+				pinctrl-0 = <&button_pins>;
+				user-event {
+					label = "User event key";
+					gpios = <&gpio 23 GPIO_ACTIVE_LOW>;
+					linux,code = <KEY_UNKNOWN>;
+				};
+			};
+		};
+	};
+
+	fragment@1 {
+		target = <&gpio>;
+		__overlay__ {
+			pinctrl-names = "default";
+
+			spi5_cs_pins {
+				/* tpm ksz*/
+				brcm,pins     = <17 18>;
+				brcm,function = <BCM2835_FSEL_GPIO_OUT>;
+				brcm,pull     = <BCM2835_PUD_OFF>;
+			};
+			spi6_cs_pins {
+				/* aout ain mbus */
+				brcm,pins     = <6 10 27>;
+				brcm,function = <BCM2835_FSEL_GPIO_OUT>;
+				brcm,pull     = <BCM2835_PUD_OFF>;
+			};
+			relais_pins: relais_pins {
+				brcm,pins     = <28>;
+				brcm,function = <BCM2835_FSEL_GPIO_OUT>;
+				brcm,pull     = <BCM2835_PUD_OFF>;
+			};
+			button_pins: button_pins {
+				brcm,pins     = <23>;
+				brcm,function = <BCM2835_FSEL_GPIO_IN>;
+				brcm,pull     = <BCM2835_PUD_OFF>;
+			};
+			tpm_pins: tpm_pins {
+				/* irq */
+				brcm,pins     = <22>;
+				brcm,function = <BCM2835_FSEL_GPIO_IN>;
+				brcm,pull     = <BCM2835_PUD_OFF>;
+			};
+			rs485_0_pins: rs485_0_pins {
+				/* tx rx rts */
+				brcm,pins     = <4 5 7>;
+				brcm,function = <BCM2835_FSEL_ALT4>;
+				brcm,pull     = <BCM2835_PUD_DOWN>;
+			};
+			rs485_1_pins: rs485_1_pins {
+				/* tx rx rts */
+				brcm,pins     = <8 9 11>;
+				brcm,function = <BCM2835_FSEL_ALT4>;
+				brcm,pull     = <BCM2835_PUD_DOWN>;
+			};
+		};
+	};
+
+	fragment@2 {
+		target = <&i2c6>;
+		__overlay__ {
+			// TODO: ANYTHING ELSE REQUIRED HERE?
+			#address-cells = <1>;
+			#size-cells = <0>;
+			pinctrl-names = "default";
+			pinctrl-0 = <&i2c6_gpio0>;
+			status = "okay";
+		};
+	};
+
+	fragment@3 {
+		target = <&i2c1>;
+		__overlay__ {
```

Comment:
still missing the `clock-frequency` property

Reply:
So what IS the frequency that we want to use? Ask the HW for this?

Reply:
First is to check what is the highest freq *all* devices on the bus can work with. Then it is pretty much the same problem as with the SPI freq.

Reply:
If the pull resistors were chosen correctly by hardware colleagues and both slaves on the bus support it (RTC and GPIO expander), then 400 kHz should be used.

On the Core and Connect we still need to use 100 kHz because of an issue with the Atmel crypto chip. (The chip goes to sleep and piserial doesn't wait for it to resume if the frequency is too high. Needs to be fixed in piserial before we can raise the frequency.)

Reply:
Yes I would ask as well, even without this comment. As I did with other places where freq. where missing or undocumented. As you said the dt describes the board. And the board defines the max freq not the chips used on the board. So the comments are rather useless.

Reply:
> If the pull resistors were chosen correctly by hardware colleagues and both slaves on the bus support it (RTC and GPIO expander), then 400 kHz should be used.
> 

According to the datasheets both device support "i2c fast mode" with 400 kHz. I will change the frequency in the DT then, thx.